apps/web/src/components/settings/RegistrationForm/RegisterFormSettings.tsx

```diff
@@ -0,0 +1,1034 @@
+import { Button } from "@/components/shadcn/ui/button";
+import { z } from "zod";
+import { RegisterFormValidator } from "@/validators/shared/RegisterForm";
+import { zodResolver } from "@hookform/resolvers/zod";
+import FormGroupWrapper from "@/components/registration/FormGroupWrapper";
+import { Checkbox } from "@/components/shadcn/ui/checkbox";
+import c, { schools, majors } from "config";
+import {
+	Command,
+	CommandEmpty,
+	CommandGroup,
+	CommandInput,
+	CommandItem,
+	CommandList,
+} from "@/components/shadcn/ui/command";
+import {
+	Popover,
+	PopoverContent,
+	PopoverTrigger,
+} from "@/components/shadcn/ui/popover";
+import { Check, ChevronsUpDown } from "lucide-react";
+import { cn } from "@/lib/utils/client/cn";
+import { useEffect, useCallback, useState } from "react";
+import { Textarea } from "@/components/shadcn/ui/textarea";
+import { FileRejection, useDropzone } from "react-dropzone";
+import { put } from "@vercel/blob";
+import { useAction } from "next-safe-action/hook";
+import {
+	modifyRegistrationData,
+	modifyResume,
+} from "@/actions/user-profile-mod";
+import { toast } from "sonner";
+import Link from "next/link";
+import { Loader2 } from "lucide-react";
+
+interface RegistrationData {
+	age: number;
+	gender: string;
+	race: string;
+	ethnicity: string;
+	wantsToReceiveMLHEmails: boolean;
+	university: string;
+	major: string;
+	shortID: string;
+	levelOfStudy: string;
+	hackathonsAttended: number;
+	softwareExperience: any;
+	heardFrom: any;
+	shirtSize: any;
+	dietRestrictions: any;
+	accommodationNote: string | null;
+	GitHub: string | null;
+	LinkedIn: string | null;
+	PersonalWebsite: any;
+	resume: string;
+}
+
+interface RegisterFormSettingsProps {
+	data: RegistrationData;
+}
+
+export default function RegisterForm({ data }: RegisterFormSettingsProps) {
+	if (data.heardFrom === null) data.heardFrom = undefined;
+	const form = useForm<z.infer<typeof RegisterFormValidator>>({
+		resolver: zodResolver(RegisterFormValidator),
+		defaultValues: {
+			hackathonsAttended: data.hackathonsAttended || 0,
+			dietaryRestrictions: data.dietRestrictions || [],
+			wantsToReceiveMLHEmails: data.wantsToReceiveMLHEmails,
+			// The rest of these are default values to prevent the controller / uncontrolled input warning from React
+			accommodationNote: data.accommodationNote || "",
+			age: data.age,
+			ethnicity: data.ethnicity as any,
+			gender: data.gender as any,
+			major: data.major,
+			github: data.GitHub as any,
+			heardAboutEvent: data.heardFrom,
+			levelOfStudy: data.levelOfStudy as any,
+			linkedin: data.LinkedIn as any,
+			personalWebsite: data.PersonalWebsite,
+			race: data.race as any,
+			shirtSize: data.shirtSize,
+			shortID: data.shortID,
+			softwareBuildingExperience: data.softwareExperience,
+			university: data.university,
+		},
+	});
+	const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+	const resumeLink: string = data.resume ?? c.noResumeProvidedURL;
+	// @ts-ignore
```

Comment:
question: Why did this you have to use `ts-ignore`?

Reply:
resumeLink thinks it could be undefined, but I thought the line above, the `??` operator will set `resumeLink` to `c.noResumeProvidedURL` if `data.resume` is nullish (including undefined, right?), so therefor, `resumeLink` will never be undefined, but I couldn't figure out how to make it think that without the build failing.

Reply:
Will take a look at this.